modules/logging/src/main/java/io/oasp/module/logging/common/impl/SecureLogging.java

```diff
@@ -0,0 +1,242 @@
+package io.oasp.module.logging.common.impl;
+
+import java.lang.reflect.Method;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.Marker;
+import org.slf4j.MarkerFactory;
+
+/**
+ * Class which provides {@link Marker}s for differential logging. Implements "MultiMarker"s for optimal filtering (if
+ * dependency org.owasp available), or corresponding conventional Markers as a fall back solution.
+ * <P>
+ * Example usage:
+ *
+ * <pre>
+ * {@code
+ * LOG.info(SecureLogging.SECURITY_FAILURE_CONFIDENTIAL, "Confidential Security Failure message.");
+ * }
+ * </pre>
+ *
+ * Example filters for appenders in logback.xml:
+ *
+ * <pre>
+ * {@code
+ * class="org.owasp.security.logging.filter.SecurityMarkerFilter"
+ * class="org.owasp.security.logging.filter.ExcludeClassifiedMarkerFilter"
+ * }
+ * </pre>
+ */
+public class SecureLogging {
+
+  /** Logger instance. */
+  private static final Logger LOG = LoggerFactory.getLogger(SecureLogging.class);
+
+  private static String extClass = "org.owasp.security.logging.SecurityMarkers";
+
+  private static String methodName = "getMarker";
+
+  private static boolean initialized = false;
+
+  private static Marker markerSecurSuccConfid = null;
+
+  private static Marker markerSecurFailConfid = null;
+
+  private static Marker markerSecurAuditConfid = null;
+
+  private static final String RESTRICTED_MARKER_NAME = "RESTRICTED";
+
+  private static final String CONFIDENTIAL_MARKER_NAME = "CONFIDENTIAL";
+
+  private static final String SECRET_MARKER_NAME = "SECRET";
+
+  private static final String TOP_SECRET_MARKER_NAME = "TOPSECRET";
+
+  private static final String SECURITY_SUCCESS_MARKER_NAME = "SECURITY SUCCESS";
+
+  private static final String SECURITY_FAILURE_MARKER_NAME = "SECURITY FAILURE";
+
+  private static final String SECURITY_AUDIT_MARKER_NAME = "SECURITY AUDIT";
+
+  // MultiMarkers by OWASP do not contain a space between the individual names, so we stick to this behavior.
+  private static final String SECURITY_SUCCESS_CONFIDENTIAL_MARKER_NAME = "SECURITY SUCCESSCONFIDENTIAL";
+
+  private static final String SECURITY_FAILURE_CONFIDENTIAL_MARKER_NAME = "SECURITY FAILURECONFIDENTIAL";
+
+  private static final String SECURITY_AUDIT_CONFIDENTIAL_MARKER_NAME = "SECURITY AUDITCONFIDENTIAL";
+
+  /**
+   * Marker for Restricted log events.
+   */
+  public static final Marker RESTRICTED = MarkerFactory.getDetachedMarker(RESTRICTED_MARKER_NAME);
+
+  /**
+   * Marker for Confidential log events. Usage with OWASP provides possibility for masking, e.g. of passwords.
+   */
+  public static final Marker CONFIDENTIAL = MarkerFactory.getDetachedMarker(CONFIDENTIAL_MARKER_NAME);
+
+  /**
+   * Marker for Secret log events.
+   */
+  public static final Marker SECRET = MarkerFactory.getDetachedMarker(SECRET_MARKER_NAME);
+
+  /**
+   * Marker for Top Secret log events.
+   */
+  public static final Marker TOP_SECRET = MarkerFactory.getDetachedMarker(TOP_SECRET_MARKER_NAME);
```

Comment:
We do not have a clear usage concept yet. However, if something is TOP_SECRET then the most secure solution is to never let it reach any logger at all. If something is present here it will be used some day (by someone who does not question this). Therefore I personally consider this Marker as misleading and would remove it.

We should in parallel open a discussion and try to collect experience in the area of Markers and esp. OWASP security markers. To finalize the entire initiative it would be great to have a more clear usage concept and some examples in the sample app (my-thai-star) in the future. Of course this last point is not a requirement for you to pick up before we can accept this PR.